documents/architectural-decisions/role-management.adoc

```diff
@@ -0,0 +1,45 @@
+== Role Management
+
+The XMPP Publish-Subscribe mechanism (XEP-0060) lacks an explicit description of how to implement role-based authentication for topics(nodes).
+It implies two possibilities on how to do so: collection nodes and the roster access model.
+
+The following aspects must be taken into account:
+
+* Preferably use existing XMPP protocols/mechanisms
+* The chosen strategy must be usable in most practical use-cases.
+
+=== Considered Options
+
+* Usage of Collection Nodes (XEP-0248)
+* Roster Access Model which supports groups (XEP-0144)
+* Custom Roles implemented on the Broker
+
+=== Decision Outcome
+
+Chosen option: Usage of Collection Nodes (XEP-0248), because the mechanism is the least intrusive that yet allows a powerful role-concept.
+We must, however, keep in mind that the XEP is deferred and might therefore not be fully supported by all XMPP servers.
```

Comment:
(Side note: Openfire does not explicitly support XEP-0248, but seems to have most features implemented, as they were part of XEP-0060 in an older version, see https://discourse.igniterealtime.org/t/xep-0248-support/75174/2)